### PR TITLE
Add functional Tests - User account log out

### PR DESCRIPTION
**Pull request by cfarhani06**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | FO - User Account : LogOut'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

### CODE REVIEW

File: tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js]:
```suggestion
```

File: tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+const homePage = require('@pages/FO/home');
+const loginPage = require('@pages/FO/login');
+const myAccountPage = require('@pages/FO/myAccount');
+
+// Import data
+const {DefaultCustomer} = require('@data/demo/customer');
+
+// Import test context
+const testContext = require('@utils/testContext');
+const baseContext = 'functional_FO_homePage_allProducts';
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js]:
```suggestion

const baseContext = 'functional_FO_homePage_allProducts';
```

File: tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    await expect(result).to.be.true;
+  });
+
+  it('should logIn', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'enterValidCredentials', baseContext);
+
+    await homePage.goToLoginPage(page);
+    await loginPage.customerLogin(page, DefaultCustomer);
+    const isCustomerConnected = await loginPage.isCustomerConnected(page);
+    await expect(isCustomerConnected, 'Customer is not connected!').to.be.true;
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js:51]:
```suggestion

    const isCustomerConnected = await loginPage.isCustomerConnected(page);
    await expect(isCustomerConnected, 'Customer is not connected!').to.be.true;

```

File: tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    const result = await homePage.isHomePage(page);
+    await expect(result).to.be.true;
+  });
+
+  it('should go to my account page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToAccountPage', baseContext);
+
+    await homePage.goToMyAccountPage(page);
+    const pageTitle = await myAccountPage.getPageTitle(page);
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js:62]:
```suggestion

    const pageTitle = await myAccountPage.getPageTitle(page);
```

File: tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    await expect(pageTitle).to.equal(myAccountPage.pageTitle);
+  });
+
+  it('should logOut with link in the footer', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'signOutWithLinkAtAccountPage', baseContext);
+
+    await myAccountPage.logout(page);
+    const isCustomerConnected = await myAccountPage.isCustomerConnected(page);
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js:71]:
```suggestion

    const isCustomerConnected = await myAccountPage.isCustomerConnected(page);
```

File: tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    await expect(isCustomerConnected, 'Customer is connected!').to.be.false;
+  });
+
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js]:
```suggestion
```

File: tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  it('should logOut with link in the footer', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'signOutWithLinkAtAccountPage', baseContext);
+
+    await myAccountPage.logout(page);
+    const isCustomerConnected = await myAccountPage.isCustomerConnected(page);
+    await expect(isCustomerConnected, 'Customer is connected!').to.be.false;
+  });
+  
+});
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js]:
```suggestion
});

```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks ;)

File: tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js (hunk 2 above)
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js]:
```suggestion
const baseContext = 'functional_FO_userAccount_logOut';
```

File: tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js (hunk 1 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js]: Done

File: tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js (hunk 3 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js:51]: Done

File: tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js (hunk 2 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js]: Done

File: tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js (hunk 4 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/FO/03_userAccount/07_logOut.js:62]: done

**Review by Progi1984 — APPROVED** [review]

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
Thanks @cfarhani06

**Pull request merged**